sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs

```diff
@@ -160,7 +160,6 @@ private void AddLoggingHeaders()
         /// </summary>
         private void AddLoggingQueryParameters()
         {
-            Diagnostics.LoggedQueryParameters.Add("api-version");
```

Comment:
I removed this because there was a Search test that validated query parameters were not duplicated. I'm not sure what the reason for this test is as there should be no problem with duplicating the parameters. I can remove the test instead if that is what is preferred. /cc @Mohit-Chakraborty @tg-msft @heaths

Reply:
I don't understand fully. This removal makes sense since this is being done in `Core`.
Why would we want to keep it for a test's sake? Is a test failing with this change?

Reply:
I removed it to fix the failing test. There is no issue with having the query param in both places but this test validates that there are no repeats - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs#L85

My point is that I don't understand the premise of this test.

Reply:
I guess it makes sense from a code hygiene perspective. I will go through and update the other usages.

Reply:
Got it. IMO if someone wrote the test to validate the behavior, I have no issues with maintaining it.
In any case, removing this from `Search` also removes duplication in code and prevents wrong deductions about which code is ensuring this parameter gets logged. So, that is a definite win.

Thanks for fixing it.

Reply:
Actually I'm surprised this change didn't break the Search recorded tests when using package refs.

Reply:
Actually, there is a slight semantic difference when removing this from the child client options classes. If someone is modifying the ClientOptions.Default static options to remove api-version, the SearchClientOptions that gets created would no longer include this, whereas if we left this in SearchClientOptions, it would still include it. I think this is probably what we want but I'm wondering if the values being set in child client options should be respecting the customer modifications to ClientOptions.Default in general.